### ISSUE TITLE
Add option for no-mixed-requires grouping with a single var

### ISSUE THREAD

**Issue by CrabDude** [eslint/eslint#2009]
First, awesome that this is a rule; I've long advocated this. However, I prefer an ordered single var/let style, and would like the same grouping enforcement but under a single let. Even the example from the [docs](http://eslint.org/docs/rules/no-mixed-requires.html) uses this style:

```javascript
var fs = require('fs'),        // "core"     \
    async = require('async'),  // "module"   |- these are "require declaration"s
    foo = require('./foo'),    // "file"     |
    bar = require(getName()),  // "computed" /
    baz = 42,                  // "other"
    bam;                       // "uninitialized"
```

There's currently no way to enforce the ordering above: [core, module, file, other, uninitialized].

**Comment by nzakas** [comment]
Can you explain what it is you'd like the rule to do? As it is, the example you have is invalid because it's missing requires with non-requires.

**Comment by CrabDude** [comment]
The feature would be an enforcement of ordering (core, module, file, computed, other, uninitialized) within a single var.
```json
{
  "rules": {
    "no-mixed-requires": [1, "onevar"]
  }
}
```

Other possible grouping option names: "ordered", "sorted", "structured"...

#### Valid:
```javascript
var fs = require('fs'),        // "core"     \
    util = require('util'),    // "core"     |
    async = require('async'),  // "module"   |- these are "require declaration"s
    _ = require('lodash'),     // "module"   |
    foo = require('./foo'),    // "file"     |
    bar = require('./bar'),    // "file"     |
    baz = require(GLOBALNAME), // "computed" |
    qux = require(getName()),  // "computed" /
    quux = 42,                 // "other"
    bim = __filename,          // "other"
    bam,                       // "uninitialized"
    bop;                       // "uninitialized"
```
```javascript
var fs = require('fs'),        // "core"
    bop;                       // "uninitialized"
```
```javascript
var foo = require('./foo'),    // "file"
    qux = require(getName()),  // "computed"
    quux = 42;                 // "other"
```

#### Invalid:
Mixed types:
```javascript
var fs = require('fs'),        // "core"
    async = require('async'),  // "module"
    util = require('util');    // "core"
```
```javascript
var fs = require('fs'),        // "core"
    util = require('util'),    // "core"
    async = require('async'),  // "module"
    foo = require('./foo'),    // "file"
    _ = require('lodash');     // "module"
```
```javascript
var fs = require('fs'),        // "core"
    quux = 42,                 // "other"
    util = require('util');    // "core"
```
Out of order:
```javascript
var foo = require('./foo'),    // "file"
    baz = require(GLOBALNAME), // "computed"
    util = require('util');    // "core"
```
```javascript
var fs = require('fs'),        // "core"
    quux = 42,                 // "other"
    foo = require('./foo');    // "file"
```
```javascript
var async = require('async'),  // "module"
    util = require('util');    // "core"
```

Though I suppose the point of eslint is that it's pluggable, so I could easily create this myself probably...

**Comment by nzakas** [comment]
Yeah, what you're describing is different than what the rule does now. The rule's goal is to ensure `require`s are always in a different `var` than other things, and it seems like you're asking for a change that allows this with an additional overlay of ordering those `require`s.

So I'm inclined to agree that this is probably better served as a custom rule than an augmentation of `no-mixed-requires`.

**Comment by gyandeeps** [comment]
Closing this as the suggestion was to make it as a custom rule.
